src/main/java/tempest/Module.java

```diff
@@ -21,21 +23,37 @@
     private UUID id;
     private String name;
     private LinkedList<StudySession> studySessions = new LinkedList<>();
+    private Color defaultColor;
+    private Color color;
 
     public Module(String name) {
         this.id = UUID.randomUUID();
         this.name = name;
+        this.defaultColor = generateDefaultColor(name);
+        this.color = defaultColor;
     }
 
     public Module(String id, String name) {
         this.id = UUID.fromString(id);
         this.name = name;
+        this.defaultColor = generateDefaultColor(name);
+        this.color = defaultColor;
     }
 
     public Module(String id, String name, LinkedList<StudySession> studySessions) {
         this.id = UUID.fromString(id);
         this.name = name;
         this.studySessions = studySessions;
+        this.defaultColor = generateDefaultColor(name);
+        this.color = defaultColor;
+    }
+
+    private Color generateDefaultColor(String name) {
+        Random rand = new Random(hash(name));
+        float r = rand.nextFloat();
+        float g = rand.nextFloat();
+        float b = rand.nextFloat();
+        return new Color(r, g, b);
```

Comment:
I was looking at having a range of colours and it is definitely possible:

https://stackoverflow.com/questions/44326765/color-mapping-for-specific-range

The only problem with this is that these sort of methods need to know the range (of modules) so setting default colour using a range would need to happen using the state class

The only reason I'm looking at this is because the random can create some really ugly colours & colours that are indistinguishable.